src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -324,19 +371,32 @@ const useCoreChatSystemHook = () => {
   React.useEffect(() => {
     if (!lastAnonymousMessage || !localDb) return;
 
-    const {type} = lastAnonymousMessage;
+    const {type, message} = lastAnonymousMessage;
     if (type === 'health.check') return;
+    if (processedMessageId === message?.id) return;
+
     if (type === 'notification.message_new') {
-      saveChannelListData(lastAnonymousMessage, ANONYMOUS).catch((e) => console.log(e));
+      // to prevent being processed twice (websocket response appears twice)
+      setProcessedMessageId(message?.id);
+      saveChannelListData(lastAnonymousMessage, ANONYMOUS)
+        .catch((e) => console.log(e))
+        .finally(() => setProcessedMessageId(null));
     }
   }, [JSON.stringify(lastAnonymousMessage), localDb]);
 
   React.useEffect(() => {
     if (!lastSignedMessage || !localDb) return;
-    const {type} = lastSignedMessage;
+
+    const {type, message} = lastSignedMessage;
     if (type === 'health.check') return;
+    if (processedMessageId === message?.id) return;
+
     if (type === 'notification.message_new' || type === 'notification.added_to_channel') {
-      saveChannelListData(lastSignedMessage, SIGNED).catch((e) => console.log(e));
+      // to prevent being processed twice (websocket response appears twice)
+      setProcessedMessageId(message?.id);
+      saveChannelListData(lastSignedMessage, SIGNED)
+        .catch((e) => console.log(e))
+        .finally(() => setProcessedMessageId(null));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In these two useEffect hooks, there's repeated code that handles the processing of messages. This violates the DRY (Don't Repeat Yourself) principle. Consider creating a separate function to handle this logic and call it in both useEffect hooks. Also, error handling should be improved beyond just logging the error to the console.


<!-- This is an auto-generated comment by OSS CodeRabbit -->